pkg/cmd/search_repo.go

```diff
@@ -199,6 +218,59 @@ func (o *searchRepoOptions) buildIndex() (*search.Index, error) {
 	return i, nil
 }
 
+func (o *searchRepoOptions) searchOCI(out io.Writer, ref string) error {
```

Comment:
OCI registries aren't really Helm repositories. Helm repositories have an `index.yaml` file. Helm repos can be searched offline. There are differences.

Helm currently has `helm search hub` and `helm search repo`. I think it would make sense to have `helm search oci` as a 3rd option and for this functionality to exist there. I could even see plugins as an option to plugin with the various providers from harbor to Docker Hub. The OCI distribution spec doesn't have a search API and there are differences out there. Plugins would be a way to deal with that variation.

What do you think?

Reply:
#i was thinking the same thing re `helm search oci`. 

good point about the different services out there requiring implementations. in helm 4 we now have structured plugin types, so we could now create a "search" plugin type (for example).

josh, re getting the latest ref - is this implementation agnostic? (sorry I can't easily check through the code right now, in transit). if so, we could perhaps have both this in helm core and plugins for per-implementation/advanced search functionality?